### PR TITLE
CI check code style

### PR DESCRIPTION
**Pull request by nicelulu**
Add check code style CI workflows.

### CODE REVIEW

**Comment by JackyWoo** [comment]
if check fails, we should print error info.

**Comment by JackyWoo** [comment]
Check failed, but shows successful.
```
/home/runner/work/RaftKeeper/RaftKeeper/src/Service/SocketReactor.cpp:291:	}
ErrorCode DECIMAL_OVERFLOW is defined but not used in file /home/runner/work/RaftKeeper/RaftKeeper/src/Common/Field.cpp
```

**Comment by nicelulu** [comment]
> Check failed, but shows successful.

Fixed see https://github.com/JDRaftKeeper/RaftKeeper/actions/runs/4467329352/jobs/7846645660

**Review by JackyWoo — CHANGES_REQUESTED** [review]

File: utils/check-style/check-whitespaces [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+
+ROOT_PATH=$(git rev-parse --show-toplevel)
+EXCLUDE_DIRS='build/|integration/|widechar_width/|glibc-compatibility/|memcpy/|consistent-hashing/|Parsers/New'
+
+# Double whitespaces
+find $ROOT_PATH/{src,base,programs,utils} -name '*.h' -or -name '*.cpp' 2>/dev/null |
+    grep -vP $EXCLUDE_DIRS |
+    while read i; do $ROOT_PATH/utils/check-style/double-whitespaces.pl < $i || echo -e "^ File $i contains double whitespaces\n"; done
```
Comment (JackyWoo) [inline utils/check-style/check-whitespaces:9]: need a blank line

**Review by JackyWoo — CHANGES_REQUESTED** [review]

File: .github/workflows/check-style.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: check-style
+
+on:
```
Comment (JackyWoo) [inline .github/workflows/check-style.yml:3]: should fix the warning

File: tests/ci/code_style_check.py [hunk 3]
```diff
@@ -109,18 +109,20 @@ def printFile(file):
...
+    subprocess.check_output(f"../../utils/check-style/check-style -n 2>&1 | tee temp/style_check/style_output.txt", shell=True)
+    subprocess.check_output(f"../../utils/check-style/check-typos 2>&1 | tee temp/style_check/typos_output.txt ", shell=True)
+    subprocess.check_output(f"../../utils/check-style/check-whitespaces -n 2>&1 | tee temp/style_check/whitespaces_output.txt ", shell=True)
+    subprocess.check_output(f"../../utils/check-style/check-duplicate-includes.sh 2>&1 | tee temp/style_check/duplicate_output.txt", shell=True)
+    subprocess.check_output(f"python3 process_style_check_result.py", shell=True)
 
     state, description, test_results, additional_files = process_result(temp_path)
 
-    if state == "failure":
+    if state == "failure" or state == "exception":
```
Comment (JackyWoo) [inline tests/ci/code_style_check.py:125]: seems not display the error message

**Review by JackyWoo — CHANGES_REQUESTED** [review]

**Pull request merged**